packages/datetime/src/dateRangeInput.tsx

```diff
@@ -356,6 +331,39 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         this.setState(nextState);
     }
 
+    private renderInputGroup = (boundary: DateRangeBoundary) => {
+        const inputProps = this.getInputProps(boundary);
+
+        // don't include `ref` in the returned HTML props, because passing it to the InputGroup
+        // leads to TS typing errors.
+        const { ref, ...htmlProps } = inputProps;
+
+        const handleInputEvent = (boundary === DateRangeBoundary.START)
+            ? this.handleStartInputEvent
+            : this.handleEndInputEvent;
+
+        const classes = classNames(inputProps.className, {
```

Comment:
always put user `className` last in the list so they can override any conditionals.

for instance, if user said `className="pt-intent-danger"` but the input was not in an error state then _that class would be removed_ by `classnames`.